pkg/cli/mirror/copy.go

```diff
@@ -612,44 +612,38 @@ func getRelatedImages(directory string, packages []v1alpha2.IncludePackage) ([]d
 	return finalList, nil
 }
 
-func isPackageSelected(bundle declcfg.Bundle, channels []declcfg.Channel, packages []v1alpha2.IncludePackage) (bool, error) {
-	isSelected := false
-	for _, pkg := range packages {
-		if pkg.Name == bundle.Package {
-			var min, max semver.Version
-			if pkg.MinVersion != "" || pkg.MaxVersion != "" {
-				version_string, err := bundleVersion(bundle)
+func isBundleSelected(bundle declcfg.Bundle, catalogChannels []declcfg.Channel, catalogFilters []v1alpha2.IncludePackage) (bool, error) {
+	var isSelected bool
+	for _, catalogFilter := range catalogFilters {
+		if catalogFilter.Name == bundle.Package {
+			if catalogFilter.MinVersion != "" || catalogFilter.MaxVersion != "" {
+				var err error
+				isSelected, err = isBundleVerSelected(catalogFilter.MinVersion, catalogFilter.MaxVersion, bundle)
 				if err != nil {
 					return isSelected, err
 				}
-				pkgVer, err := semver.Make(version_string)
-				if err != nil {
-					return isSelected, err
-				}
-				if err != nil {
-					return isSelected, err
-				}
-				if pkg.MinVersion != "" {
-					min, err = semver.Make(pkg.MinVersion)
-					if err != nil {
-						return isSelected, err
-					}
-				}
-				if pkg.MaxVersion != "" {
-					max, err = semver.Make(pkg.MaxVersion)
-					if err != nil {
-						return isSelected, err
+			} else if len(catalogFilter.Channels) > 0 {
+				for _, channelFilter := range catalogFilter.Channels {
```

Comment:
Also lets brainstorm this,  cognitive complexity  is high and lets look at ways to simplify

Reply:
One thing I want to mention here is that I don't any dependency resolution when the bundle is selected. Apologies if I missed that. I am wondering if there could be some code reuse from this package https://github.com/openshift/oc-mirror/blob/365db76579ca829c56a0ab026588a52f0d13d084/pkg/operator/diff/internal/diff.go#L319.

Reply:
Hi @jpower432, according to our conversation on January 30, we are going to set skipDependencies as true since it is going to be deprecated in future.